apps/service_orchestrate/test/orchestrate/event/handler_test.exs

```diff
@@ -172,4 +172,37 @@ defmodule Orchestrate.Event.HandlerTest do
       assert nil == Orchestrate.Schedule.Store.get!(schedule.dataset_id, schedule.subset_id)
     end
   end
+
+  test "deletes job on definition:delete", %{schedule: schedule} do
+    Orchestrate.Scheduler.new_job()
+    |> Job.set_name(:"#{schedule.dataset_id}__#{schedule.subset_id}")
+    |> Job.set_schedule(Crontab.CronExpression.Parser.parse!("* * * * *"))
+    |> Job.set_task({IO, :puts, ["hello"]})
+    |> Orchestrate.Scheduler.add_job()
+
+    Brook.Test.with_event(@instance, fn ->
+      Orchestrate.Schedule.Store.persist(schedule)
+    end)
+
+    delete = %Delete{id: "123", dataset_id: schedule.dataset_id, subset_id: schedule.subset_id}
+
+    Brook.Test.send(@instance, definition_delete(), "testing", delete)
+
+    assert_async do
+      assert nil ==
+               Orchestrate.Scheduler.find_job(:"#{schedule.dataset_id}__#{schedule.subset_id}")
+    end
+
+    # We should be testing this but are not due to JSR
```

Comment:
Why is the JSR not there? (Also, is it worth leaving the comment hanging around?)

Reply:
I knew someone was going to ask, and now I've forgotten everything but the pain. Let me see if I can remember why.